services/proxy/pkg/middleware/oidc_auth.go

```diff
@@ -271,13 +226,45 @@ func (m *oidcAuth) getProvider() OIDCProvider {
 		// provider needs to be cached as when it is created
 		// it will fetch the keys from the issuer using the .well-known
 		// endpoint
-		provider, err := m.providerFunc()
+		provider, err := m.ProviderFunc()
 		if err != nil {
-			m.logger.Error().Err(err).Msg("could not initialize oidcAuth provider")
+			m.Logger.Error().Err(err).Msg("could not initialize oidcAuth provider")
 			return nil
 		}
 
 		m.provider = provider
 	}
 	return m.provider
 }
+
+func (m OIDCAuthenticator) Authenticate(r *http.Request) (*http.Request, bool) {
+	// there is no bearer token on the request,
+	if !m.shouldServe(r) {
```

Comment:
This does not need to check for `isPublicPath` like the `BasicAuthenticator` does?

Reply:
Good point, it should. :+1: